CRM/Core/BAO/Block.php

```diff
@@ -239,153 +245,134 @@ public static function create($blockName, &$params, $entity = NULL, $contactId =
       );
     }
     else {
-      $contactId = $params['contact_id'];
+      $contact_id = $params['contact_id'];
     }
 
-    $updateBlankLocInfo = CRM_Utils_Array::value('updateBlankLocInfo', $params, FALSE);
+    // Get current and submitted values
+    $existing_values = self::getBlockIds($blockName, $contact_id, $entityElements, $updateBlankLocInfo);
+    $submitted_values = $params[$blockName];
 
-    //get existsing block ids.
-    $blockIds = self::getBlockIds($blockName, $contactId, $entityElements, $updateBlankLocInfo);
-
-    if (!$updateBlankLocInfo) {
-      $resetPrimaryId = NULL;
-      $primaryId = FALSE;
-      foreach ($params[$blockName] as $count => $value) {
-        $blockId = CRM_Utils_Array::value('id', $value);
-        if ($blockId) {
-          if (is_array($blockIds)
-            && array_key_exists($blockId, $blockIds)
-          ) {
-            unset($blockIds[$blockId]);
-          }
-          else {
-            unset($value['id']);
-          }
-        }
-        //lets allow to update primary w/ more cleanly.
-        if (!$resetPrimaryId && !empty($value['is_primary'])) {
-          $primaryId = TRUE;
-          if (is_array($blockIds)) {
-            foreach ($blockIds as $blockId => $blockValue) {
-              if (!empty($blockValue['is_primary'])) {
-                $resetPrimaryId = $blockId;
-                break;
-              }
-            }
-          }
-          if ($resetPrimaryId) {
-            $baoString = 'CRM_Core_BAO_' . $blockName;
-            $block = new $baoString();
+    // For each submitted value
+    foreach ($submitted_values as $count => $submitted_value) {
```

Comment:
To be fair, I don't think Civi's coding standards take a hard line on this. But it's true that we use camelCase far more often than underscores and consistency is nice.